### PR TITLE
Sync players and mobs over network

### PR DESCRIPTION
**Pull request by Ratoone**
Instantiate and sync both the players and the enemies over network. It should still be possible to play single player as well

### CODE REVIEW

**Comment by Ratoone** [comment]
Do not merge before #3

File: Assets/Scripts/Player/PlayerMovement.cs [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
     void Update() {
+        if (PhotonNetwork.IsConnected && !photonView.IsMine) {
+            return;
+        }
+
         movement.x = Input.GetAxisRaw("Horizontal");
         movement.y = Input.GetAxisRaw("Vertical");
         movement = movement.normalized;
 
         animator.SetFloat("Speed", movement.sqrMagnitude);
```
Comment (GianiStatie) [inline Assets/Scripts/Player/PlayerMovement.cs:31]: The animation does not update for the remote player

**Comment by GianiStatie** [comment]
We need to create a sorting order for environments, because the spawn platform renders on top of the player. I'll create a task

File: Assets/Scripts/Manager/PlayerInstanceManager.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    private List<GameObject> spawnPositions;
+
+    // Start is called before the first frame update
+    void Start() {
+        if (spawnPositions.Count <= 0) {
+            return;
+        }
+
+        if (PhotonNetwork.IsConnected) {
+            PhotonNetwork.Instantiate(this.playerPrefab.name, spawnPositions[PhotonNetwork.LocalPlayer.ActorNumber - 1].transform.position, Quaternion.identity);
```
Comment (GianiStatie) [inline Assets/Scripts/Manager/PlayerInstanceManager.cs]: If the maximum number of players is 2 for a room, why so many spawn platforms? We can randomly spawn the player or have players choose where to spawn (each spawn point leading to more powerful monsters)

**Comment by Ratoone** [comment]
Made a comment in the task, will add fixes

**Review by GianiStatie — APPROVED** [review]

File: Assets/Scripts/Manager/PlayerInstanceManager.cs (hunk 2 above)
Comment (Ratoone) [reply Assets/Scripts/Manager/PlayerInstanceManager.cs]: The maximum number i was thinking to be 6. Forgot to change that part

Comment (Ratoone) [reply Assets/Scripts/Manager/PlayerInstanceManager.cs]: I think it's better to have static spawning points. Like portals? Something that the player comes out of

Comment (GianiStatie) [reply Assets/Scripts/Manager/PlayerInstanceManager.cs]: we can leave them as is, don't need to make it look fancier (except if Karina really wants to make the art)

**Pull request merged**